File: C++/1930. Unique Length-3 Palindromic Subsequences.cpp
class Solution {
public:
    int countPalindromicSubsequence(string s) {
        map<int,int>a;
        int c=0;
        for(int i=0;i<s.size()-2;i++){
            if(a[s[i]]!=-1){
            for(int j=s.size()-1;j>1;j--){
                if(s[i]==s[j]){
                    a[s[i]]=-1;
                    set<int>x;
                    for(int k=i+1;k<j;k++)
                    x.insert(s[k]);
                    c+=x.size();
                    break;
                }
            }
            }
        }
        return c;
    }
};

File: JAVA/1930. Unique Length-3 Palindromic Subsequences.java
class Solution {
    public int countPalindromicSubsequence(String s) {
        Map<Character, Integer> a = new HashMap<>();
        int c = 0;
        
        for (int i = 0; i < s.length() - 2; i++) {
            if (!a.containsKey(s.charAt(i))) {
                for (int j = s.length() - 1; j > 1; j--) {
                    if (s.charAt(i) == s.charAt(j)) {
                        a.put(s.charAt(i), -1);
                        HashSet<Character> x = new HashSet<>();
                        for (int k = i + 1; k < j; k++) {
                            x.add(s.charAt(k));
                        }
                        c += x.size();
                        break;
                    }
                }
            }
        }
        
        return c;
    }
}